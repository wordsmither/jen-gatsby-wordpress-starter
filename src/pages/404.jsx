import React from 'react'
import { graphql } from 'gatsby'

import Layout from '../components/layout'
import Seo from '../components/seo'

const NotFoundPage = ({ data, location }) => {
	const siteTitle = data.site.siteMetadata.title

	return (
		<Layout location={location} title={siteTitle}>
			<div className="flex flex-col mx-auto text-xl prose text-center mt-12">
				<h2>404: Could not find the requested page</h2>
			</div>
		</Layout>
	)
}

export default NotFoundPage

export function Head() {
	return <Seo title="404: Not Found" />
}

export const pageQuery = graphql`
	query {
		site {
			siteMetadata {
				title
			}
		}
	}
`
