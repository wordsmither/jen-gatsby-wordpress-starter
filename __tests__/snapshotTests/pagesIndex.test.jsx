import { render } from '@testing-library/react'

import PageIndexTemplate from '../../src/templates/pagesIndex'
import pages from '../data/pages.json'

describe('<PageIndexTemplate />', () => {
	beforeEach(() => {
		location = {
			pathname: '/pages/1',
			state: { breakOpen: false },
		}
	})
	it('should render with pages', () => {
		const { asFragment } = render(
			<PageIndexTemplate
				pageContext={{ itemsPerPage: 2, routing: true, pages }}
				location={location}
			/>,
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
