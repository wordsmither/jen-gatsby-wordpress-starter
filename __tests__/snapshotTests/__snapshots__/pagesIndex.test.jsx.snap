// Vitest Snapshot v1

exports[`<PageIndexTemplate /> > should render with pages 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen max-h-screen min-w-screen max-w-screen flex flex-col"
  >
    <div
      class="my-0 pt-10 px-5 text-xl"
      data-is-root-path="true"
    >
      <nav>
        <ul
          class="flex flex-row flex-wrap sm:flex-nowrap list-none justify-between max-w-screen-sm mx-auto"
        >
          <li
            class="mr-auto"
          >
            <a
              class="font-sans hover:underline"
              href="/"
            >
              üè† Home
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/posts"
            >
              üì∞ Posts
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/pages"
            >
              üìë Pages
            </a>
          </li>
          <li
            class="mx-4"
          >
            <a
              class="font-sans hover:underline"
              href="/examples"
            >
              ‚öõÔ∏è Examples
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <main
      class="mb-auto"
    >
      <header
        class="prose text-2xl mx-auto mt-20"
      >
        <h1
          class="text-center mx-auto"
        >
          Pages
        </h1>
      </header>
      <div
        class="max-w-screen-lg mx-auto"
      >
        <section>
          <div
            class="max-w-full"
          >
            <span
              class="mb-8 prose-sm font-bold"
            >
              Page 1/1
            </span>
            <section>
              <div
                class="mt-12 grid gap-5 max-w-content mx-auto lg:grid-cols-3 lg:max-w-screen-lg"
              >
                <a
                  href="/pages/test-page/"
                >
                  <div
                    class="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500"
                  >
                    <div
                      class="flex-shrink-0 relative h-40"
                    >
                      <div
                        class="w-full h-full bg-gradient-to-b from-blue-100 to-blue-500"
                      />
                    </div>
                    <h2
                      class="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900"
                    >
                      Test Page ‚Üí
                    </h2>
                  </div>
                </a>
                <a
                  href="/pages/sample-page/"
                >
                  <div
                    class="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500"
                  >
                    <div
                      class="flex-shrink-0 relative h-40"
                    >
                      <div
                        class="w-full h-full bg-gradient-to-b from-blue-100 to-blue-500"
                      />
                    </div>
                    <h2
                      class="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900"
                    >
                      Sample Page ‚Üí
                    </h2>
                  </div>
                </a>
              </div>
            </section>
            <div
              class="sticky lg:bottom-12 bottom-4 mt-10"
            />
          </div>
        </section>
      </div>
    </main>
    <footer
      class="w-full text-white bg-black p-4 mt-12"
    >
      <nav
        class="flex flex-col max-w-lg mx-auto lg:max-w-screen-lg"
      >
        <ul>
          <li
            class="ml-3 list-disc text-blue-300 hover:text-blue-100"
          >
            <a
              class="text-blue-300 hover:text-blue-100 focus:text-purple-600 active:text-purple-300 "
              href="/posts/example-post-with-image/"
            >
              Example Post with Image
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="flex my-4 p-2"
      >
        <span
          class="mx-auto"
        >
          ¬© 2023 Built with 
          <a
            class="text-purple-300 underline hover:text-blue-100"
            href="https://www.gatsbyjs.com"
          >
            Gatsby
          </a>
           and 
          <a
            class="text-blue-200 underline hover:text-blue-100"
            href="https://wordpress.org/"
          >
            WordPress
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
